firewood/src/merkle.rs

```diff
@@ -2323,7 +2323,6 @@ mod tests {
     }
 
     #[test]
-    #[ignore]
```

Comment:
Code was approved but there was a suggestion to add tests. I added tests but couldn't get them to pass. The runtime would crash when attempting to drop the parent node (in the case of these tests, the sentinel). I verified that this behaviour "expected" on `main` prior to merging the ignored tests, using the `write_node!` macro (how we were updating nodes before I introduced a regression). 

To be perfectly honest, I don't quite understand why there was a panic while dropping the sentinel. I spent a couple of hours trying to figure it out to no avail. In any case dropping a node with `inner == None` should never be an issue since there's nothing to actually put in the cache. 

Put another way, I don't quite know what the `insert` code does differently that allows it to work on `main` while the tests fail.